src/le.py

```diff
@@ -3325,6 +3415,107 @@ def cmd_follow(args):
+
+    arg = args[0]
+    pname = os.path.abspath(arg)
+    pnames = glob.glob(pname)
+    if len(pnames) == 0:
+        die('\nWarning: No Files found for %s\n' % pname)
+    files_with_log_names = get_log_names_for_files(pnames)
+    # user input required before new files are set to be followed
+    files_to_follow = user_prompt(files_with_log_names)
+    debug
+    for file in files_to_follow:
+        print "request to follow %s " % file
+    ### AT THIS TIME IGNORING THE LIST OF FILES RETURNED
+    ### file_to_follow AS NEED TO CONFIRM AGENT BEHAVIOUR
+    name = config.name
+    if name == NOT_SET:
+        name = os.path.basename(pname)
+    type_opt = config.type_opt
+    if type_opt == NOT_SET:
+        type_opt = ""
+    filename = "Multiple:" + pname
+    request_follow(filename, name, type_opt)
+
+def user_prompt(files_and_log_names):
+    """
+    Displays 2 lists - files already followed with log names, and those not.
+    Prompts user if they wish to follow files or not
+    """
+    ### NOTE THAT THIS WILL NEED REVIEW AND EXPECT TO CHANGE
+    ### THE OUTPUT WAS MADE UP ON THE SPOT AND THE EXPECTED
+    ### OPTIONS MAY CHANGE AS WELL AS THE TEXT
+    ### THIS USER PROMPT MAY OVER COMPLICATED THINGS
+
+    files_already_followed = {}
+    files_not_followed = {}
+    for filename, logname in files_and_log_names.items():
+        #if debug here
+        #print "filename: %s and logname: %s" % (filename, logname)
+        if logname is NOT_SET:
+            files_not_followed[filename] = NOT_SET
+        else:
+            files_already_followed[filename] = logname
+    num_already_followed = len(files_already_followed)
+    num_not_followed =  len(files_not_followed)
+    print "\nFiles found already being followed, Total %s:" % num_already_followed
+    # print files being followed if any
+    if num_already_followed == 0:
+        print "\t\n No files found that are being followed."
+    else:
+        print "LOGNAME \t FILENAME"
+        for filename, logname in files_and_log_names.items():
+            if logname is not None:
+                print "%s \t %s" % (logname, filename)
+    # print files not be being followed if any
+    if num_not_followed == 0:
+        print "\nAll files found are already being followed\n"
+        sys.exit(EXIT_OK)
+    else:
+        print "\nFiles found not being followed, Total %s:" % num_not_followed
+        print "\n".join(files_not_followed)
+    # prompt user with options to follow files
+    while True:
+        print "\nSelect one of the follow options: 0, 1 or 2"
+        print "0 quit"
+        print "1 Follow only the %s files not currently being followed?" % num_not_followed
+        print "2 Assign all files found, being followed or not, to the one log"
+        user_resp = raw_input(':')
+        try:
+            user_resp = int(user_resp)
+        except ValueError:
+            print "Please try again: 0, 1, or 2"
+        else:
+            if user_resp == 0:
```

Comment:
yes - totally agree - thing is this whole section may change - but I agree with what you are saying and will keep it in mind